syncer/checkpoint.go

```diff
@@ -553,6 +556,43 @@ func (cp *RemoteCheckPoint) FlushPointsExcept(tctx *tcontext.Context, exceptTabl
 	return nil
 }
 
+// FlushPointWithTableInfo implements CheckPoint.FlushPointWithTableInfo
+func (cp *RemoteCheckPoint) FlushPointWithTableInfo(tctx *tcontext.Context, schema string, table string, ti *model.TableInfo) error {
+	cp.RLock()
+	defer cp.RUnlock()
+
+	sqls := make([]string, 0, 10)
+	args := make([][]interface{}, 0, 10)
+
+	point := cp.points[schema][table]
+
+	tiBytes, err := json.Marshal(ti)
+	if err != nil {
+		return terror.ErrSchemaTrackerCannotSerialize.Delegate(err, schema, table)
+	}
+
+	location := point.MySQLLocation()
+	sql2, arg := cp.genUpdateSQL(schema, table, location, nil, tiBytes, false)
+	sqls = append(sqls, sql2)
+	args = append(args, arg)
+
+	// use a new context apart from syncer, to make sure when syncer call `cancel` checkpoint could update
+	tctx2, cancel := tctx.WithContext(context.Background()).WithTimeout(maxDMLConnectionDuration)
```

Comment:
I suddently realized if this action cost a long time, other checkpoint method maybe blocked. Hope it will not block upward to some user interface commands

Reply:
Since operate-schema can only use when task paused, the impact should be minimal